indico/modules/categories/client/js/calendar.js

```diff
@@ -105,7 +105,20 @@ import CalendarLegend from './components/CalendarLegend';
       },
       events({start, end}, successCallback, failureCallback) {
         function updateCalendar(data) {
-          const attr = groupBy === 'category' ? 'categoryId' : 'venueId';
+          let attr;
+          switch (groupBy) {
+            case 'category':
+              attr = 'categoryId';
+              break;
+            case 'location':
+              attr = 'venueId';
+              break;
+            case 'room':
+              attr = 'roomId';
+              break;
+            default:
+              throw new Error(`Invalid "groupBy": ${groupBy}`);
+          }
```

Comment:
```js
const attr = {category: 'categoryId', location: 'venueId', room: 'roomId'}[groupBy];
```

(if you want error handling: `if(!attr) {throw ...}`)

Reply:
Fixed!